# Chapter 1

Data: https://www.kaggle.com/datasets/arjunprasadsarkhel/2021-olympics-in-tokyo

In [13]:
import pandas as pd
import numpy as np
import scipy.stats as stats

In [6]:
medal_count = pd.read_excel("Data/2021 Olympics in Tokyo/Medals.xlsx")
medal_count

/Users/zigapetelin/anaconda3/lib/python3.11/site-packages/openpyxl/styles/stylesheet.py:226: UserWarning: Workbook contains no default style, apply openpyxl's default
  warn("Workbook contains no default style, apply openpyxl's default")


Rank                    Team/NOC  Gold  Silver  Bronze  Total  \
0      1    United States of America    39      41      33    113   
1      2  People's Republic of China    38      32      18     88   
2      3                       Japan    27      14      17     58   
3      4               Great Britain    22      21      22     65   
4      5                         ROC    20      28      23     71   
..   ...                         ...   ...     ...     ...    ...   
88    86                       Ghana     0       0       1      1   
89    86                     Grenada     0       0       1      1   
90    86                      Kuwait     0       0       1      1   
91    86         Republic of Moldova     0       0       1      1   
92    86        Syrian Arab Republic     0       0       1      1   

    Rank by Total  
0               1  
1               2  
2               5  
3               4  
4               3  
..            ...  
88             77  
89             77  
90             77  
91             77  
92             77  

[93 rows x 7 columns]

### Estimates of Location

In [9]:
#Mean
medal_count["Total"].mean()
print("On average, a country got", medal_count["Total"].mean(), "medals")

On average, a country got 11.612903225806452 medals


In [17]:
# Trimmed mean and Median
print(stats.trim_mean(medal_count["Total"], 0.1))
print(medal_count["Total"].median())

6.8933333333333335
4.0


We can see the trimmed mean, which trims the tails (extremes) of the distributions, is closer to the median compared to the mean. This might indicate fat tails.

In [19]:
np.average(medal_count["Total"], weights=medal_count["Gold"])

46.832352941176474

Weighing medal count by the number of gold medals moves the average to 46.8. This means that those that got more medals in total, also got more gold medals.

In [23]:
print(medal_count["Gold"].mean())
print(stats.trim_mean(medal_count["Gold"], 0.1))

3.6559139784946235
1.96


Trimming 10% from each side of gold medals tells us that on average each country got 1.96 gold medals. The difference from the normal average tells us that we again, probably have fat tails (a few countries with a lot of gold medals).

In [25]:
#Weighted median def from the book
def weighted_median(df, median_col, weight_col):
    df_sorted = df.sort_values(median_col)
    cumsum = df_sorted[weight_col].cumsum()
    cutoff = df_sorted[weight_col].sum() / 2.
    return df_sorted[cumsum >= cutoff][median_col].iloc[0]

weighted_median(medal_count, "Total", "Gold")

40

In [30]:
q3 = np.quantile(medal_count["Total"], 0.75)
#75% have less than 11 medals and 25% more than 11.

In [29]:
medal_count.describe()

Rank       Gold     Silver     Bronze       Total  Rank by Total
count  93.000000  93.000000  93.000000  93.000000   93.000000      93.000000
mean   46.333333   3.655914   3.634409   4.322581   11.612903      43.494624
std    26.219116   7.022471   6.626339   6.210372   19.091332      24.171769
min     1.000000   0.000000   0.000000   0.000000    1.000000       1.000000
25%    24.000000   0.000000   0.000000   1.000000    2.000000      23.000000
50%    46.000000   1.000000   1.000000   2.000000    4.000000      47.000000
75%    70.000000   3.000000   4.000000   5.000000   11.000000      66.000000
max    86.000000  39.000000  41.000000  33.000000  113.000000      77.000000

### Estimates of Variability